app/src/pages/addBidPage/addBid.jsx

```diff
@@ -0,0 +1,257 @@
+        if (file) {
+            reader.onload = (e) => {
+                result = e.target.result
+                imageData = [result, file.name]
+
+                setImagePortfolios((prev) => {
+                    const temp = [...prev];
+                    temp[currentIndex] = [...new Set([...temp[currentIndex], [result, file.name]])]
+                    return temp;
+                })
+
+            }
+            reader.readAsDataURL(file)
+
+        }
+    }
+
+    const changeDescriptionNext = () => {
+        if (currentIndex < 2) {
+            setCurrentIndex((prev) => {
+                return prev+1}
+            )
+        }
+    }
+
+    const handlePost = async () => {
+        if (description){
+            const allImageData = []
+
+            for (let i in imagePortfolios){
+                const imageData = []
+
+                for (let image in imagePortfolios[i]){
+                    imageData.push({
+                        "name" : imagePortfolios[i][image][1],
+                        "blob" : imagePortfolios[i][image][0]
+                    })
+                }
+                allImageData.push(imageData)
+            }
+
+            await API.addBid(allImageData, description, postId, userId)
+        }
+
+        setNavHome(true)
+
+    }
+
+
+    return (
+        <div id={styles.addBid}>
+            <Header />
+
+            {
+                (navHome) &&
+                (<Navigate to="/" />)
+            }
+
+            <div id={styles.post}>
+
+                <div id={styles.images} >
+                    {
+                        (postData.images) &&
+                        (<GenImage image={postData.images[currentIndex]} postPage={true} style={"main"} />)
+                    }
+
+                    <div id={styles.imageBtns}>
+
+                        { imageBtns.map((index) => {
+                            // go through post Images
+                            return(
+                                <FontAwesomeIcon icon={faCircle} color={ currentIndex==index ? "grey" : "rgb(216, 216, 216)"}
+                                    onClick={(e)=> {
+                                        setCurrentIndex(index)
+                                    }}
+                                />
+                            )
+                        })}
+                    </div>
+                </div>
+
+
+                <div id={styles.postDetails}>
+                    <h1>{postData.title}</h1>
+                    <h4>{postData.category}</h4>
+                    <p>{postData.description}</p>
+                </div>
+
+
+            </div>
+
+            <div id={styles.bid}
+                style={{ display: finish? "none" : "flex" }}
+            >
+                <h1>Portfolio Images </h1>
+
+                <div id={styles.addImages} onClick={() =>{fileInputRef.current?.click()}} >
+                    <input type="file" ref={fileInputRef} style={{ display: "none" }}
```

Comment:
I remember you brought this up in our 1:1. Creative way of using useRef to have your own custom button using the native file uploader!